### PR TITLE
S13/cart page/feature/left side

### PR DESCRIPTION
**Pull request by vynnykpff**
**Check list**

- UI
- Validations

### CODE REVIEW

File: src/components/Cart/components/CartOrderDetails/components/CartConfirmationInfo/components/CartConfirmationInfoAgreement/CartAgreementList.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  const handleToggleChecked = (itemId: string) => {
+    setCheckedItems(prev => {
+      if (prev.includes(itemId)) {
+        return prev.filter(id => id !== itemId);
+      } else {
+        return [...prev, itemId];
+      }
+    });
+
+    if (checkedItems.length >= 2) {
```
Comment (Yuliia143) [inline src/components/Cart/components/CartOrderDetails/components/CartConfirmationInfo/components/CartConfirmationInfoAgreement/CartAgreementList.tsx]: Magic number

File: src/components/UI/CartPhoneInput/CartPhoneInput.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      setSelectedCountryCode("");
+    }
+  }, [fields.country]);
+
+  const getCountryCode = () => {
+    const filteredCountries = countries.filter(country => {
+      return country.name.toLowerCase() === fields.country.toLowerCase();
+    });
+
+    if (filteredCountries.length > 0) {
```
Comment (Yuliia143) [inline src/components/UI/CartPhoneInput/CartPhoneInput.tsx]: Simplify ?

File: src/components/UI/CartPhoneInput/CartPhoneInput.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+  const handleLabelClick = () => {
+    setIsPhoneInputFocused(true);
+  };
+
+  const handlePhoneBlur = (e: FocusEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    setIsValidNumber(isValidPhoneNumber(value));
+
+    if (isValidPhoneNumber(value)) {
```
Comment (Yuliia143) [inline src/components/UI/CartPhoneInput/CartPhoneInput.tsx]: In this way you call it twice

File: src/components/UI/CartTextarea/CartTextarea.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  const {
+    fields: { orderNotes },
+  } = useAppSelector(state => state.cart);
+  const { setField } = useActions();
+  const [textareaValue, setTextareaValue] = useState(orderNotes);
+
+  const [debouncedTextareaValue] = useDebounce(textareaValue, GlobalDelay.DEFAULT);
+
+  useChangeEffect(() => {
+    setField({ key: "orderNotes", value: textareaValue });
```
Comment (Yuliia143) [inline src/components/UI/CartTextarea/CartTextarea.tsx]: It's better to have constants/enums for fields if you want to pass it

File: src/utils/getCartLocation.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  cities: string[];
+};
+
+export const getCartLocation = ({ fieldName, countries, states, cities }: Params) => {
+  if (fieldName === "country") {
+    return countries.map(country => country.name);
+  }
+
+  if (fieldName === "city") {
+    return cities;
```
Comment (Yuliia143) [inline src/utils/getCartLocation.ts]: The same - field name condition

**Review by Yuliia143 — APPROVED** [review]

**Review by Yuliia143 — APPROVED** [review]

**Review by NikitaIASA — APPROVED** [review]

**Pull request merged**